components/common/src/test/java/com/hotels/styx/config/validator/DocumentFormatTest.java

```diff
@@ -443,86 +445,127 @@ public void validatesSubobjects() throws Exception {
         assertThat(result, is(true));
     }
 
+
     @Test
-    public void reusesObjectSchemas() throws Exception {
-        JsonNode rootObject = YAML_MAPPER.readTree(""
-                + "httpPipeline: \n"
-                + "  pipeline: 'interceptorsList'\n"
-                + "  handler: 'handlerName'\n"
-                + "plugins:\n"
-                + "  all: 'x, y z'\n");
+    public void routingObjectDefinition() throws Exception {
+
+        Schema.FieldValue proxyTo = object(
+                field("id", string()),
+                field("destination", string())
+        );
+
+        Schema.FieldValue redirection = object(
+                field("status", integer()),
+                field("location", string())
+        );
 
         DocumentFormat validator = newDocument()
-                .subSchema("HttpPipeline", schema(
-                        field("pipeline", string()),
-                        field("handler", string())
-                ))
-                .subSchema("PluginsList", schema(
-                        field("all", string())
-                ))
-                .rootSchema(schema(
-                        field("httpPipeline", object("HttpPipeline")),
-                        field("plugins", object("PluginsList"))
+                .typeExtension("ProxyTo", proxyTo)
+                .typeExtension("Redirection", redirection)
+                .rootSchema(object(
+                        field("httpPipeline", new Schema.RoutingObjectSpec())
                 ))
                 .build();
 
-        assertThat(validator.validateObject(rootObject), is(true));
+        boolean outcome1 = validator.validateObject(
```

Comment:
These variable names do not describe the difference in scenarios being tested. Should there be two different test methods instead?

Reply:
Refactored. The variable names are no longer there.